File: drive-kotlin/src/main/kotlin/xyz/block/ftl/drive/verb/hotswap/FtlHotswapAgentPlugin.java
package xyz.block.ftl.drive.verb.hotswap;

import org.hotswap.agent.annotation.LoadEvent;
import org.hotswap.agent.annotation.OnClassLoadEvent;
import org.hotswap.agent.annotation.Plugin;
import org.hotswap.agent.javassist.CtClass;
import xyz.block.ftl.drive.Logging;

@Plugin(name = "FtlHotswapAgentPlugin", testedVersions = {})
public class FtlHotswapAgentPlugin {
  @OnClassLoadEvent(classNameRegexp = ".*", events = LoadEvent.REDEFINE)
  public static void loaded(CtClass ctClass) {
    Logging.Companion.logger(FtlHotswapAgentPlugin.class.getName())
        .info("Reloaded " + ctClass.getName());
  }
}

File: kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/hotswap/FtlHotswapAgentPlugin.kt
package xyz.block.ftl.hotswap

import org.hotswap.agent.annotation.LoadEvent
import org.hotswap.agent.annotation.OnClassLoadEvent
import org.hotswap.agent.annotation.Plugin
import org.hotswap.agent.javassist.CtClass
import xyz.block.ftl.logging.Logging

@Plugin(name = "FtlHotswapAgentPlugin", testedVersions = [])
object FtlHotswapAgentPlugin {
  private val logger = Logging.logger(FtlHotswapAgentPlugin::class)

  @JvmStatic
  @OnClassLoadEvent(classNameRegexp = ".*", events = [LoadEvent.REDEFINE])
  fun loaded(ctClass: CtClass) {
    logger.debug("Reloaded " + ctClass.name)
  }
}
